### ISSUE TITLE
Unexpected warning for pack indexing

### ISSUE THREAD

**Issue by breese** [llvm/llvm-project#112931]
Suppose we use the feature test macro to only use pack indexing when available
```
template <std::size_t N, typename... Args>
constexpr auto my_element(Args... args) {
#if __cpp_pack_indexing >=  202311L
  return args...[N];
#else
  // Do something else
#endif
}
```
A compiler warning is emitted when compiling for C++23 or earlier
```
<source>:13:16: warning: pack indexing is a C++2c extension [-Wc++26-extensions]
   13 |         return args...[N];
      |                ^
1 warning generated.
```

**Comment by zwuis** [comment]
Similar issue: #109311.

**Comment by cor3ntin** [comment]
right, the feature test macro is set in all language modes because the feature is supported as an extension in all language modes
